plugin/trino-elasticsearch/src/test/java/io/trino/plugin/elasticsearch/BaseElasticsearchConnectorTest.java

```diff
@@ -1269,6 +1269,61 @@ public void testBoolean()
         assertThat(rows.getMaterializedRows()).containsExactlyInAnyOrderElementsOf(expected.getMaterializedRows());
     }
 
+    @Test
+    public void testCustomDateFormat()
+            throws IOException
+    {
+        String indexName = "custom_date_format";
+        @Language("JSON")
+        String mapping = "" +
+                "{" +
+                "  \"properties\": { " +
+                "    \"ts1\":{\"type\": \"date\", \"format\": \"epoch_millis\"}," +
+                "    \"ts2\":{\"type\": \"date\", \"format\": \"epoch_second\"}," +
+                "    \"ts3\":{\"type\": \"date\", \"format\": \"strict_date_optional_time\"}," +
+                "    \"ts4\":{\"type\": \"date\", \"format\": \"yyyy-MM-dd'T'HH:mm:ss.SSSZZ\"}," +
+                "    \"ts5\":{\"type\": \"date\", \"format\": \"yyyy-MM-dd HH:mm:ss\"}," +
+                "    \"ts6\":{\"type\": \"date\", \"format\": \"yyyy-MM-dd HH:mm:ss||epoch_second\"}" +
+                "  }" +
+                "}";
+        createIndex(indexName, mapping);
+        index(indexName, ImmutableMap.<String, Object>builder()
+                .put("ts1", 1667820637001L)
+                .put("ts2", 1667820637L)
+                .put("ts3", "2022-11-07T11:30:37.000+0000")
+                .put("ts4", "2022-11-07T19:30:37.001+0800")
+                .put("ts5", "2022-11-07 11:30:37")
+                .put("ts6", 1667820637L)
+                .buildOrThrow());
+
+        assertQuery(
```

Comment:
`assertQuery` is legacy. Use `assertThat(query(...))`. See other examples in this file.